modules/setup/library/Setup/WebWizard.php

```diff
@@ -711,7 +709,23 @@ public function getRequirements($skipModules = false)
                 'setup.requirement.class'
             )
         )));
-        $set->merge($pgsqlSet);
+        $dbSet->merge($pgsqlSet);
+        $set->merge($dbSet);
+
+        $dbRequire = new RequirementSet(true);
+        $dbRequire->add(new SetRequirement(array(
+            'optional'      => $dbSet->getState(),
+            'condition'     => $dbSet->getState(),
```

Comment:
Please pass the actual set here to the requirement. It's a *Set*Requirement after all, not a *Boolean*Requirement.